dali/test/python/test_operator_arithmetic_ops.py

```diff
@@ -186,89 +199,111 @@ def get_numpy_input(input, kind, type):
             device_id = 0)
     pipe.build()
     pipe_out = pipe.run()
-    l = as_cpu(pipe_out[0]).as_array()
-    r = as_cpu(pipe_out[1]).as_array()
-    out = as_cpu(pipe_out[2]).as_array()
-    assert_equals(out.dtype, target_type)
-    l_np = get_numpy_input(l, left_kind, target_type)
-    r_np = get_numpy_input(r, right_kind, target_type)
-    if 'f' in np.dtype(target_type).kind:
-        np.testing.assert_allclose(out, op(l_np, r_np),
-            rtol=1e-07 if target_type != np.float16 else 0.005)
-    else:
-        np.testing.assert_array_equal(out, op(l_np, r_np))
+    for sample in range(batch_size):
+        l = as_cpu(pipe_out[0]).at(sample)
+        r = as_cpu(pipe_out[1]).at(sample)
+        out = as_cpu(pipe_out[2]).at(sample)
+        assert_equals(out.dtype, target_type)
+        l_np = get_numpy_input(l, left_kind, target_type)
+        r_np = get_numpy_input(r, right_kind, target_type)
+        if 'f' in np.dtype(target_type).kind:
+            np.testing.assert_allclose(out, op(l_np, r_np),
+                rtol=1e-07 if target_type != np.float16 else 0.005)
+        else:
+            np.testing.assert_array_equal(out, op(l_np, r_np))
+
+
+def test_arithmetic_ops_big():
+    for kinds in input_kinds:
+        for (op, op_desc) in sane_operations:
+            for types_in in [(np.int8, np.int8)]:
+                yield check_arithm_op, kinds, types_in, op, shape_big, op_desc
 
 def test_arithmetic_ops():
     for kinds in input_kinds:
         for (op, op_desc) in sane_operations:
             for types_in in itertools.product(input_types, input_types):
-                yield check_arithm_op, kinds, types_in, op, op_desc
-
+                yield check_arithm_op, kinds, types_in, op, shape_small, op_desc
 
 # The div operator that always returns floating point values
-def check_arithm_fdiv(kinds, types):
+def check_arithm_fdiv(kinds, types, shape):
     left_type, right_type = types
     left_kind, right_kind = kinds
     target_type = div_promote(left_type, right_type)
-    iterator = iter(ExternalInputIterator(batch_size, left_type, right_type))
+    iterator = iter(ExternalInputIterator(batch_size, shape, left_type, right_type, left_kind, right_kind))
     pipe = ExprOpPipeline(kinds, types, iterator, (lambda x, y : x / y), batch_size = batch_size,
             num_threads = 2, device_id = 0)
     pipe.build()
     pipe_out = pipe.run()
-    l = as_cpu(pipe_out[0]).as_array()
-    r = as_cpu(pipe_out[1]).as_array()
-    out = as_cpu(pipe_out[2]).as_array()
-    assert_equals(out.dtype, target_type)
-    l_np = get_numpy_input(l, left_kind, target_type)
-    r_np = get_numpy_input(r, right_kind, target_type)
-    np.testing.assert_allclose(out, l_np / r_np,
-        rtol=1e-07 if target_type != np.float16 else 0.005)
+    for sample in range(batch_size):
+        l = as_cpu(pipe_out[0]).at(sample)
+        r = as_cpu(pipe_out[1]).at(sample)
```

Comment:
extract the comparison code to a shared utility function (maybe even put it in test_utils.py if you think it's appropriate) and use it here

Reply:
done